### PR TITLE
Coveo Initial Commit

### PR DESCRIPTION
**Pull request by Delinea-RickRoca**
### 🛠 Changes being made
Initial Coveo Release

# Coveo Delinea Secret Server Integration

This package is designed to discover Coveo User Accounts. It will also provide a method of determining Administrative and Service accounts based on a configurable criteria. It will provide detailed instructions and the necessary Scripts to perform these functions.

### 🏎 Quality check

- [X] Should this be added to Delinea Marketplace?

- [X] Has this been tested and verified?

- [X] Have you included a separate .md documentation for each script submitted?

### CODE REVIEW

File: Scripts/SecretServer/Coveo/Discovery/Coveo Discovery.ps1 [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+            "0" { [string]$MessageLevel = "INF0 " }
+            "1" { [string]$MessageLevel = "WARN " }
+            "2" { [string]$MessageLevel = "ERROR" }
+            "3" { [string]$MessageLevel = "DEBUG" }
+        }
+        # Write Log data
+        $MessageString = "{0}`t| {1}`t| {2}`t| {3}" -f $Timestamp, $MessageLevel,$logApplicationHeader, $Message
+        $MessageString | Out-File -FilePath $LogFile -Encoding utf8 -Append -ErrorAction SilentlyContinue
+        # $Color = @{ 0 = 'Green'; 1 = 'Cyan'; 2 = 'Yellow'; 3 = 'Red'}
+        # Write-Host -ForegroundColor $Color[$ErrorLevel] -Object ( $DateTime + $Message)
```
Comment (tylerezimmerman) [inline Scripts/SecretServer/Coveo/Discovery/Coveo Discovery.ps1:53]: @Delinea-RickRoca why is this comment out block left here?

File: Scripts/SecretServer/Coveo/Discovery/readme.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
+# Coveo Account Discovery
+Add Disclaimer
+
+## Create Discovery Source
+.
```
Comment (tylerezimmerman) [inline Scripts/SecretServer/Coveo/Discovery/readme.md:5]: Remove this "."

File: Scripts/SecretServer/Coveo/Discovery/readme.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    - This completes the creation of the Saas Scan Template Creation
+ 
+
+### Create Account Scan Template
+
+- Log in to Secret Server Tenant
+- Navigate to **ADMIN** > **Discovery** > **Configuration** >   **Scanner Definition** > **Scan Templates** 
+- Click **Create Scan Template**
+- Fill out the required fields with the information
+    - **Name:** (Evxample: Coveo Account)
```
Comment (tylerezimmerman) [inline Scripts/SecretServer/Coveo/Discovery/readme.md:31]: Fix spelling of "Example"

File: Scripts/SecretServer/Coveo/Discovery/readme.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    - **Active:** (Checked)
+    - **Scan Type:** Account
+    - **Parent Scan Template:** Account(Basic)
+    - **Fields**
+        - Change Resource to **tenant-url**
+        - Add field: Admin-Account (Leave Parent and Include in Match Blank)
+        - Add field: Service-Account (Leave Parent and Include in Match Blank)
+        - Add field: Local-Account (Leave Parent and Include in Match Blank)
+    - Click Save
+    - This completes the creation of the Account Scan Template Creation
```
Comment (tylerezimmerman) [inline Scripts/SecretServer/Coveo/Discovery/readme.md:41]: This should be moved left once space to align with "Fill out the required fields with the information"

File: Scripts/SecretServer/Coveo/Discovery/readme.md [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+ 
+### Create Discovery Script
+
+- Log in to Secret Server Tenant
+- Navigate to**ADMIN** > **Scripts**
```
Comment (tylerezimmerman) [inline Scripts/SecretServer/Coveo/Discovery/readme.md:46]: Add space between to and ADMIN

File: Scripts/SecretServer/Coveo/Instructions.md [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+​
+### Prerequisites
+
+- Access to a Coveo instance with administrative privileges.
+API Key with the the required privileges.
+
+## Create an OAuth Application Registry
+
+- Have, or create, a new API Key that has an Account Administrator role permission.  
+-
```
Comment (tylerezimmerman) [inline Scripts/SecretServer/Coveo/Instructions.md:28]: Remove duplicate "-"

File: Scripts/SecretServer/Coveo/RemotePasswordChanger/readme.md
Comment (tylerezimmerman) [inline Scripts/SecretServer/Coveo/RemotePasswordChanger/readme.md:1]: Remove references to Juniper Mist

File: Scripts/SecretServer/Coveo/Templates/readme.md
Comment (tylerezimmerman) [inline Scripts/SecretServer/Coveo/Templates/readme.md:1]: Remove references to Juniper Mist

**Review by tylerezimmerman — CHANGES_REQUESTED** [review]

File: Scripts/SecretServer/Coveo/Discovery/Coveo Discovery.ps1 (hunk 1 above)
Comment (Delinea-RickRoca) [reply Scripts/SecretServer/Coveo/Discovery/Coveo Discovery.ps1:53]: Used for debugging when running scripts manually. Not used in automated Logging.  Will ad comments to indicate that in futre releases and will make a note to go back and review once we hit the 34

**Review by tylerezimmerman — APPROVED** [review]

**Pull request merged**